src/os/ui/list.js

```diff
@@ -209,17 +210,32 @@ os.ui.Module.directive('list', [os.ui.listDirective]);
  */
 os.ui.listLink = function(scope, element, attr, ctrl) {
   // add the list stuff to the scope
-  var id = attr['listId'];
+  const id = attr['listId'];
   ctrl.id = id;
 
-  var list = os.ui.list.get(id) || scope.$eval(attr['listItems']);
+  const list = os.ui.list.get(id) || scope.$eval(attr['listItems']);
 
   if (list) {
     ctrl.items = /** @type {!Array<!os.ui.list.ListEntry>} */ (list);
   }
 
   ctrl.prefix = attr['listPrefix'];
   ctrl.suffix = attr['listSuffix'];
+
+  Object.keys(attr).filter((key) => key.match(/^listAttr.+$/))
+      .reduce((acc, key) => {
+        let camel = key.replace(/^listAttr/, '');
+        camel = camel.charAt(0).toLowerCase() + camel.substr(1);
+        ctrl.attrs[camel] = scope.$eval(attr[key]);
+        scope.$watch(function() {
+          const kebab = key.replace(/([A-Z])/g, '-$1').toLowerCase();
+          return scope.$eval(element.attr(kebab));
+        }, function(newValue, oldValue) {
+          ctrl.attrs[camel] = newValue;
+          ctrl.update_();
```

Comment:
this looks like it re-compiles the directive anytime any of the parameters change

Reply:
It now only compiles when !item.element, which is the first time it sees it.